src/asset-swapper/quote_consumers/exchange_proxy_swap_quote_consumer.ts

```diff
@@ -343,6 +345,46 @@ export class ExchangeProxySwapQuoteConsumer implements SwapQuoteConsumerBase {
             };
         }
 
+        // OTC orders
+        if (
+            [ChainId.Mainnet, ChainId.PolygonMumbai].includes(this.chainId) && // @todo goerli and polygon?
+            quote.orders.every((o) => o.type === FillQuoteTransformerOrderType.Otc) &&
+            !requiresTransformERC20(optsWithDefaults)
+        ) {
+            const otcOrdersData = quote.orders.map((o) => o.fillData as NativeOtcOrderFillData);
+            const fillAmountPerOrder = generateFillAmounts(sellAmount, quote);
```

Comment:
Looks like `fillAmountPerOrder` won't be used unless `quote.order.length === 1`.
I'm little confused why this is needed. Could we use takerAmount directly?